src/compas/files/_iotools.py

```diff
@@ -0,0 +1,71 @@
+"""For the time being, these functions are only for internal use."""
+from contextlib import contextmanager
+import io
+
+try:
+    from urllib.request import urlopen
+except ImportError:
+    from urllib2 import urlopen
+
+
+@contextmanager
+def open_file(file_or_filename, mode='r'):
+    """Context-manager to open a file, path or URL and return a corresponding file object.
+
+    If the argument is a path-like object, it will open it,
+    if the argument is an HTTP(s) URL, it will open it, and
+    if the argument is already a file-like object, it will simply yield it.
+
+    This context manager ensures that when using a path-like object as input,
+    the file is closed once it it goes out of scope.
```

Comment:
I think the fact that a file-like object is not closed by this context manager should really be underscored in the docs.

Reply:
added a note about that on https://github.com/compas-dev/compas/pull/712/commits/0f707368d373675e0c6b7fcf646c7f5812038d5d